pkg/observability/common.go

```diff
@@ -14,8 +14,95 @@
 
 package observability
 
+import (
+	"context"
+	"time"
+
+	"go.opencensus.io/stats"
+	"go.opencensus.io/tag"
+)
+
+// This file defines some common variables used by both the key server and
+// verification server.
+
+var (
+	// BlameTagKey indicating Who to blame for the API request failure.
+	// NONE: no failure
+	// CLIENT: the client is at fault (e.g. invalid request)
+	// SERVER: the server is at fault
+	// EXTERNAL: some third party is at fault
+	// UNKNOWN: for everything else
+	BlameTagKey = tag.MustNewKey("blame")
+
+	// ResultTagKey contains a free format text describing the result of the
+	// request. Preferably ALL CAPS WITH UNDERSCORE.
+	// OK indicating a successful request.
+	// You can losely base this string on
+	// https://github.com/googleapis/googleapis/blob/master/google/rpc/code.proto
+	// but feel free to use any text as long as it's easy to filter.
+	ResultTagKey = tag.MustNewKey("result")
+
+	// buildIDTagKey is the tag key for the build_id.
+	buildIDTagKey = tag.MustNewKey("build_id")
+
+	// buildTagTagKey is the tag key for the build_tag.
+	buildTagTagKey = tag.MustNewKey("build_tag")
+)
+
+var (
+	// BlameNone indicate no API failure
+	BlameNone = tag.Upsert(BlameTagKey, "NONE")
+
+	// BlameClient indicate the client is at fault (e.g. invalid request)
+	BlameClient = tag.Upsert(BlameTagKey, "CLIENT")
+
+	// BlameServer indicate the server is at fault
+	BlameServer = tag.Upsert(BlameTagKey, "SERVER")
+
+	// BlameExternal indicate some third party is at fault
+	BlameExternal = tag.Upsert(BlameTagKey, "EXTERNAL")
+
+	// BlameUnknown can be used for everything else
+	BlameUnknown = tag.Upsert(BlameTagKey, "UNKNOWN")
+)
+
+var (
+	// ResultOK add a tag indicating the API call is a success.
+	ResultOK = tag.Upsert(ResultTagKey, "OK")
+	// ResultNotOK add a tag indicating the API call is a failure.
+	ResultNotOK = ResultError("NOT_OK")
+)
+
+var (
```

Comment:
I think a lot of this is already defined in https://github.com/google/exposure-notifications-server/blob/main/pkg/observability/observability.go. We may want to combine/reconcile the two files.

Reply:
https://github.com/google/exposure-notifications-server/blob/main/pkg/observability/observability.go#L37

Reply:
Yes, those are removed.